compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -2249,21 +2252,76 @@ public void visit(BLangRecordLiteral recordLiteral) {
                 BLangExpression spreadOpExpr = spreadOpField.expr;
 
                 analyzeExpr(spreadOpExpr);
-                if (spreadOpExpr.type.tag != TypeTags.RECORD) {
+
+                BType spreadOpExprType = spreadOpExpr.type;
+                int spreadFieldTypeTag = spreadOpExprType.tag;
+                if (spreadFieldTypeTag == TypeTags.MAP) {
+                    if (inclusiveTypeSpreadField != null) {
+                        this.dlog.error(spreadOpExpr.pos, DiagnosticCode.MULTIPLE_INCLUSIVE_TYPES);
+                        break;
+                    } else {
+                        inclusiveTypeSpreadField = spreadOpField;
+                    }
+                    if (fields.size() > 1) {
+                        if (names.size() > 0) {
+                            this.dlog.error(spreadOpExpr.pos, DiagnosticCode.INCLUSIVE_MAPPING_TYPE_IN_SPREAD_FIELD,
+                                spreadOpExpr);
+                            break;
+                        }
+                        // Skipping to avoid multiple error messages
+                        continue;
+                    }
+                }
+
+                if (spreadFieldTypeTag != TypeTags.RECORD) {
                     continue;
                 }
 
-                for (BField bField : ((BRecordType) spreadOpExpr.type).fields.values()) {
-                    if (Symbols.isOptional(bField.symbol)) {
-                        continue;
+                BRecordType spreadExprRecordType = (BRecordType) spreadOpExprType;
+                boolean isSpreadExprRecordTypeSealed = spreadExprRecordType.sealed;
+                if (!isSpreadExprRecordTypeSealed) {
+                    // More than one inclusive-type-descriptors with spread-fields are not allowed.
+                    if (inclusiveTypeSpreadField != null) {
+                        this.dlog.error(spreadOpExpr.pos, DiagnosticCode.MULTIPLE_INCLUSIVE_TYPES);
+                        break;
+                    } else {
+                        inclusiveTypeSpreadField = spreadOpField;
                     }
+                }
 
+                LinkedHashMap<String, BField> fieldsInRecordType = ((BRecordType) spreadOpExpr.type).fields;
+                for (Object fieldName : names) {
+                    if (fieldsInRecordType.containsKey(fieldName)) {
+                        if (fieldsInRecordType.get(fieldName).type.tag != TypeTags.NEVER) {
+                            // An error is logged if the same field is occurred, and it is not defined as a never-type
+                            this.dlog.error(spreadOpExpr.pos, DiagnosticCode.DUPLICATE_KEY_IN_RECORD_LITERAL_SPREAD_OP,
+                                    recordLiteral.expectedType.getKind().typeName(), fieldName, spreadOpField);
+                        }
+                    } else {
+                        if (!isSpreadExprRecordTypeSealed) {
+                            // Log an error if the spread-field has a inclusive type
```

Comment:
Do we need this comment now that the check is kinda self-explanatory?

Reply:
Removed